.github/workflows/pullrequest-closed.yml

```diff
@@ -19,21 +19,78 @@ jobs:
           tag: pr-${{ github.event.pull_request.number }}
           untagged-older-than: 28
 
-  # If a PR is closed it should no longer lock any testservers
-  remove-testserver-locks:
+  # If a PR is closed the testserver lock should be removed and corresponding badges updated
+  process_labels:
+    name: Process labels
     runs-on: ubuntu-latest
+    outputs:
+      labels: ${{ steps.process.outputs.labels }}
+      badges: ${{ steps.process.outputs.badges }}
     steps:
-      - uses: actions-ecosystem/action-remove-labels@v1
+      - name: Process labels
+        id: process
+        uses: actions/github-script@v6
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          labels: |
-            lock:artemis-test1
-            lock:artemis-test2
-            lock:artemis-test3
-            lock:artemis-test4
-            lock:artemis-test5
-            lock:artemis-test6
-            lock:artemis-test7
-            lock:artemis-test8
-            lock:artemis-test9
-            lock:artemis-test10
+          script: |
+              const labelsToRemove = [];
+              const labelsToProcess = [];
+
+              // Get the PR number
+              const prNumber = context.payload.pull_request.number;
+
+              // Iterate through labels on the PR
+              for (const label of context.payload.pull_request.labels) {
+                const labelName = label.name;
+                const regex = /^lock:artemis-test(\d+)$/;
+
+                if (regex.test(labelName)) {
+                  // Extract the part after "lock:" using capture groups
+                  const extractedLabel = labelName.match(regex)[1];
+                  labelsToProcess.push(extractedLabel);
+                  labelsToRemove.push(labelName);
+                }
+              }
+
+              // Do something with the extracted labels
+              console.log('Badges to process:', labelsToProcess);
+              console.log('Labels to remove:', labelsToRemove);
+
```

Comment:
Dev comment?

Reply:
I left these intentionally for logging what happens here, in case something goes wrong in the future.